src/shell/shell_cmds.cpp

```diff
@@ -608,6 +614,61 @@ static std::vector<int> calc_column_widths(const std::vector<int> &word_widths,
 	return col_widths;
 }
 
+char buffer[15] = {0};
+char *format_date(uint16_t year, uint8_t month, uint8_t day)
+{
+	char formatstring[6];
+	char date_format = dos.tables.country[0];
+	char date_separator = dos.tables.country[11];
+	sprintf(formatstring,
+	        date_format == 1 ? "D%cM%cY"
+	                         : (date_format == 2 ? "Y%cM%cD" : "M%cD%cY"),
+	        date_separator, date_separator);
+	Bitu bufferptr = 0;
```

Comment:
1. Please use `size_t` for array indexes and `uint64_t` for big numeric counters or large integer math (or timer counters).  (we always want to drop or avoid Bitu` and move to the circa-1999 C-standard types: https://en.cppreference.com/w/c/types/integer)
2.  Suggest calling this `offset` or `index`, because `ptr` implies it's a pointer.

Reply:
Done.